### PR TITLE
Make sure the application is shutdown properly in the shutdown hook

### LINKED ISSUE #11792: Command line app ignores SIGINT (Ctrl+C)
**Describe the bug**
Quarkus command line apps do not respond to SIGINT

**Expected behavior**
A Quarkus command line app should initiate shutdown when passed SIGINT (Ctrl+C on a terminal)

**Actual behavior**
The signal is ignored, nothing happens

**To Reproduce**
1. Clone https://github.com/ivangreene/my-quarkus-cli
2. Run `mvn clean package`
3. Run `java -jar target/my-quarkus-cli-1.0-SNAPSHOT-runner.jar`
4. Press Ctrl+C

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `Darwin ivan.local 19.6.0 Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64 i386 MacBookPro16,1 Darwin`
 - Output of `java -version`:
```
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: `1.7.1.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/Cellar/maven/3.6.3_1/libexec
Java version: 11.0.7, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.6", arch: "x86_64", family: "mac"
```

### PR DESCRIPTION
**Pull request by gastaldi**
Also make sure the specified exit handler is invoked during SIGINT

Fixes #11792

### CODE REVIEW

File: core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java [hunk 1]
```diff
@@ -294,6 +294,12 @@ public void run() {
             } finally {
                 stateLock.unlock();
             }
+            if (currentApplication.isStarted()) {
+                // On CLI apps, SIGINT won't call io.quarkus.runtime.Application#stop(),
+                // making the awaitShutdown() below block the application termination process
+                // It should be a noop if called twice anyway
```
Comment (gastaldi) [inline core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java:300]: This was what I observed with the reproducer project in the issue

**Comment by geoand** [comment]
Seems reasonable, but this is one is up to Stuart

**Comment by gsmet** [comment]
Assigning Stuart so that he can have a look when he's back from PTO.

File: core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java [hunk 2]
```diff
@@ -159,36 +159,36 @@ public static void run(Application application, Class<? extends QuarkusApplicati
...
         };
-        handleSignal("INT", handler);
-        handleSignal("TERM", handler);
+        handleSignal("INT", exitHandler);
+        handleSignal("TERM", exitHandler);
         // the HUP and QUIT signals are not defined for the Windows OpenJDK implementation:
         // https://hg.openjdk.java.net/jdk8u/jdk8u-dev/hotspot/file/7d5c800dae75/src/os/windows/vm/jvm_windows.cpp
-        if (OS.getCurrent() == OS.WINDOWS) {
-            handleSignal("BREAK", quitHandler);
+        if (OS.WINDOWS.isCurrent()) {
```
Comment (gsmet) [inline core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java:187]: It was there before but I'm not sure it's a good idea to use an util coming from GraalVM svm artifact for such a low level thing.

Comment (gastaldi) [reply core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java]: Changed to use `System.getProperty("os.name")`. We need an utility to reuse that in various places

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**